pygmt/tests/test_figure.py

```diff
@@ -116,6 +116,7 @@ def test_figure_savefig_filename_with_spaces():
     fig.basemap(region=[0, 1, 0, 1], projection="X1c/1c", frame=True)
     with GMTTempFile(prefix="pygmt-filename with spaces", suffix=".png") as imgfile:
         fig.savefig(imgfile.name)
+        assert r"\040" not in os.path.abspath(imgfile.name)
         assert os.path.exists(imgfile.name)
```

Comment:
A problem with this PR is that running `fig.savefig` with filenames that have spaces (introduced in #1116) will produce an output like "pygmt-filename\040with\040spaces.png". This test currently passes still, but it is a **false positive**, because Python's `os.path.exists` reads "pygmt-filename\040with\040spaces.png" as ""pygmt-filename with spaces.png"".

Any ideas how to improve the test and/or implementation of `psconvert`?